src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -955,6 +955,64 @@ public void EndWrite(IAsyncResult asyncResult)
 #endif
         }
 
+        // ReadAsync - provide async read functionality.
+        // 
+        // This method provides async read functionality. All we do is
+        // call through to the Begin/EndRead methods.
+        // 
+        // Input:
+        // 
+        //     buffer            - Buffer to read into.
+        //     offset            - Offset into the buffer where we're to read.
+        //     size              - Number of bytes to read.
+        //     cancellationtoken - Token used to request cancellation of the operation
+        // 
+        // Returns:
+        // 
+        //     A Task<int> representing the read.
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int size, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.CanBeCanceled)
```

Comment:
I don't understand this part: if the `cancellationToken` is [capable of being canceled](https://msdn.microsoft.com/en-us/library/system.threading.cancellationtoken.canbecanceled(v=vs.110).aspx) then return a canceled task? Shouldn't this check be `cancellationToken.IsCancellationRequested` instead?

What if the `cancellationToken` is not in fact canceled?